ArduCopter/Parameters.cpp

```diff
@@ -1224,6 +1224,34 @@ const AP_Param::GroupInfo ParametersG2::var_info2[] = {
     AP_GROUPINFO("TKOFF_RPM_MAX", 7, ParametersG2, takeoff_rpm_max, 0),
 #endif
 
+    // @Param: GCS_MODE_BLOCK
```

Comment:
Maybe we could call the parameter FLTMODE_DISABLE(D) and then apply it to all situations?  E.g. just completely block the vehicle from being able to enter these modes.

Reply:
I think all situations is a bad idea and I don't really see the benefit. There are no other paths to accidentally change to the wrong mode. Via RC requires you to have setup the modes you want on the switch, scripting requires you to have added a script that does the wrong thing. Then that just leaves all the various failsafe reasons for changing modes, I can't see any case where blocking a failsafe mode change is a good idea. 

TLDR: blocking all modes changes leaves the possibility for conflicting parameters, EG `RCx_OPTION` set to mode loiter but loiter disabled.

Reply:
I agree with Peter, from use case perspective. This is to prevent pilot mistakes. There is not other path where it would be a mistake.
Flight modes configured on RC will all be intentional and planned to use. Otherwise it would not be configured on the RC. Then only other ways to change modes is mission or fail-safes and scripts. 
Modes setup on FS or scripts we would not want to block obviously.

Reply:
let's call it FLTMODE_GCSBLOCK